New Location and MiniProfileCard components

maxme opened the pull request:
Closes #88

![Screenshot_20240408_082511](https://github.com/Automattic/Gravatar-SDK-Android/assets/40213/156a4d77-bea4-45d3-bb7d-77ea5374ae10)

### Description

New:
- Mini profile component (added to the demo app)
- Location component (which is just an ExpandableText using the profile location)

### Testing Steps

- Run the demo app
- Check the Profile tab
- Make sure the card is matching the design

maxme committed:
New Location and MiniProfileCard components

hamorillo commented on gravatar-ui/src/main/java/com/gravatar/ui/components/MiniProfileCard.kt:
```diff
@@ -0,0 +1,75 @@
+
+@Composable
+fun MiniProfileCard(profile: UserProfile, modifier: Modifier = Modifier) {
+    Column(
+        modifier = modifier,
+        horizontalAlignment = Alignment.Start,
+        verticalArrangement = Arrangement.Top,
+    ) {
+        Row {
```
⚠️  We don't need the column or the horizontal/vertical alignments.

```suggestion
    Row(modifier = modifier) {
```

maxme replied:
Updated in ef36d1370ffed46828e611176de890930d7497d1

hamorillo commented on gravatar-ui/src/main/java/com/gravatar/ui/components/MiniProfileCard.kt:
```diff
@@ -0,0 +1,75 @@
+                size = 72.dp,
+                modifier = Modifier.clip(CircleShape),
+            )
+            Column(modifier = Modifier.padding(14.dp, 0.dp, 0.dp, 0.dp)) {
```
⛏
```suggestion
            Column(modifier = Modifier.padding(start = 14.dp)) {
```

Should we also add the padding at the end?

```suggestion
            Column(modifier = Modifier.padding(horizontal = 14.dp)) {
```

maxme replied:
> Should we also add the padding at the end?

I'd keep the start padding only, it seems easier to manage in a card like in there:

<img width="383" alt="Screenshot 2024-04-08 at 14 56 53" src="https://github.com/Automattic/Gravatar-SDK-Android/assets/40213/5ed5eb54-5437-4049-876d-fbc2d3accd24">



(24dp padding start and end on the card)

hamorillo commented on demo-app/src/main/java/com/gravatar/demoapp/ui/DemoGravatarApp.kt:
```diff
@@ -210,6 +211,14 @@ private fun ProfileTab(modifier: Modifier = Modifier, onError: (String?, Throwab
+                    Modifier
+                        .background(MaterialTheme.colorScheme.surfaceContainer)
+                        .fillMaxWidth()
+                        .padding(24.dp, 24.dp, 24.dp, 24.dp),
```
⛏
```suggestion
                        .padding(24.dp),
```

maxme replied:
Updated in ef36d1370ffed46828e611176de890930d7497d1

hamorillo commented:
Looks nice! I left some minor comments, the more interesting could be [this one](https://github.com/Automattic/Gravatar-SDK-Android/pull/93#discussion_r1555497639).

maxme committed:
Remove unecessary layout node and update padding notation

Co-authored-by: Héctor Abraham Morillo Prieto <hector.abraham.morillo.prieto@automattic.com>

maxme commented:
I updated the code according to your comments and I added a "maxLines" option to the location component, also changed the default to 1 line as it seems more appropriate for a location only.

hamorillo reviewed (approved):
:shipit:
